dev/docker.sh

```diff
@@ -72,14 +37,77 @@ if [[ -z $DOCKER_PASSWORD ]]; then
 	exit 1
 fi
 
+VERSION="$TRAVIS_TAG"
+VERSION_SHORT=$( echo $VERSION | cut -d. -f1,2 )
+
+if [[ -z $VERSION ]]; then
+	echo "empty VERSION"
+	exit 1
+fi
+
+if [[ -z $VERSION_SHORT ]]; then
+	echo "empty VERSION_SHORT"
+	exit 1
+fi
+
+# Build the image.
+docker build \
+    -t $IMAGE:$VERSION \
+    -t $IMAGE:$VERSION_SHORT \
+    -t $IMAGE:latest .
+
+#
+# Run the image in container. This is not strictly needed however
+# serves as additional test in automatic builds.
+#
+docker run -d $IMAGE
+docker ps -a
+
 # Publish the image to Docker hub.
 if [ -n "$DOCKER_PASSWORD" -a -n "$DOCKER_USERNAME" -a -n "$VERSION" ]; then
-	echo "Logging into docker"
+	echo "Logging into Docker Hub"
 	echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
 
 	# All the tags need to be pushed individually:
 	for tag in $VERSION $VERSION_SHORT latest; do
-		echo "Pushing docker image for tag $tag"
-		docker push opengrok/docker:$tag
+		echo "Pushing Docker image for tag $tag"
+		docker push $IMAGE:$tag
 	done
 fi
+
+# Update README file in Docker hub.
+push_readme() {
+	declare -r image="${1}"
+	declare -r token="${2}"
+	declare -r input_file="${3}"
+
+	if [[ ! -r $input_file ]]; then
+		echo "file $input_file is not readable"
+		exit 1
+	fi
+
+	local code=$(jq -n --arg msg "$(<${input_file})" \
+	    '{"registry":"registry-1.docker.io","full_description": $msg }' | \
+	        curl -s -o /dev/null  -L -w "%{http_code}" \
+	           https://cloud.docker.com/v2/repositories/"${image}"/ \
+	           -d @- -X PATCH \
+	           -H "Content-Type: application/json" \
+	           -H "Authorization: JWT ${token}")
```

Comment:
just to understand it...this reads the input file (=README) and sends it through the api to docker, right?

Reply:
looks like nice oneliner but little hard to follow at first sight

Reply:
yes. I copied the whole thing from elsewhere.